apps/judicial-system/backend/src/app/modules/defendant/limitedAccessDefendant.controller.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  JwtAuthGuard,
+  RolesGuard,
+  RolesRules,
+  CurrentHttpUser,
+} from '@island.is/judicial-system/auth'
+import { ServiceRequirement, User } from '@island.is/judicial-system/types'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import {
+  Controller,
+  UseGuards,
+  Inject,
+  Patch,
+  Param,
+  Body,
+} from '@nestjs/common'
+import { ApiTags, ApiOkResponse } from '@nestjs/swagger'
+import { prisonSystemStaffRule } from '../../guards'
+import { CaseExistsGuard, CurrentCase, Case } from '../case'
+import { DefendantService } from './defendant.service'
+import { UpdateDefendantDto } from './dto/updateDefendant.dto'
+import { CurrentDefendant } from './guards/defendant.decorator'
+import { DefendantExistsGuard } from './guards/defendantExists.guard'
+import { Defendant } from './models/defendant.model'
+import type { Logger } from '@island.is/logging'
+import { prisonSystemStaffUpdateRule } from './guards/roleRules'
+
+interface LimitedAccessUpdateDefendant
+  extends Pick<UpdateDefendantDto, 'punishmentType'> {}
+
+@Controller('api/case/:caseId/limitedAccess/defendant')
+@ApiTags('limited access defendant')
+@UseGuards(JwtAuthGuard, RolesGuard)
+export class LimitedAccessDefendantController {
+  constructor(
+    private readonly defendantService: DefendantService,
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+  ) {}
+
+  @UseGuards(CaseExistsGuard, DefendantExistsGuard)
+  @RolesRules(prisonSystemStaffUpdateRule)
```

Comment:
This will ensure that prison staff can only pass in `punishmentType` and no other fields of the `UpdateDefendantDto`

Reply:
I only added tests that check the Roles rules config but not a test that enforces the logic it applies.